pageserver/src/tenant/layer_map.rs

```diff
@@ -417,18 +433,21 @@ where
     /// TODO The optimal number should probably be slightly higher than 1, but to
     ///      implement that we need to plumb a lot more context into this function
     ///      than just the current partition_range.
-    pub fn is_reimage_worthy(layer: &L, partition_range: &Range<Key>) -> bool {
+    pub fn is_reimage_worthy(layer: &L, partition_range: &Range<Key>) -> Result<bool> {
```

Comment:
nit: It never returns `Err`. Is there a reason for this change?

Reply:
`overlaps` may return error